tardis/model.py

```diff
@@ -250,116 +250,48 @@ def save_spectra(self, fname):
-        include_from_model_in_hdf5 = {'plasma_array': include_from_plasma_, 'j_blues': None,
-                                      'runner': include_from_runner_,
-                                      'last_line_interaction_in_id': None,
-                                      'last_line_interaction_out_id': None,
-                                      'last_line_interaction_shell_id': None, 'montecarlo_nu': None,
-                                      'luminosity_density': _save_luminosity_density,
-                                      'luminosity_density_virtual': _save_spectrum_virtual,
-                                      'configuration_dict': _save_configuration_dict,
-                                      'last_line_interaction_angstrom': None}
-
-        if isinstance(buffer_or_fname, basestring):
-            hdf_store = pd.HDFStore(buffer_or_fname)
-        elif isinstance(buffer_or_fname, pd.HDFStore):
-            hdf_store = buffer_or_fname
-        else:
-            raise IOError('Please specify either a filename or an HDFStore')
-        logger.info('Writing to path %s', path)
-
-        def _get_hdf5_path(path, property_name):
-            return os.path.join(path, property_name)
-
-        def _to_smallest_pandas(object):
-            try:
-                return pd.Series(object)
-            except Exception:
-                return pd.DataFrame(object)
-
-
-        def _save_model_property(object, property_name, path, hdf_store):
-            property_path = _get_hdf5_path(path, property_name)
-
-            try:
-                object.to_hdf(hdf_store, property_path)
-            except AttributeError:
-                _to_smallest_pandas(object).to_hdf(hdf_store, property_path)
-
-
-        for key in include_from_model_in_hdf5:
-            if include_from_model_in_hdf5[key] is None:
-                _save_model_property(getattr(self, key), key, path, hdf_store)
-            elif callable(include_from_model_in_hdf5[key]):
-                include_from_model_in_hdf5[key](key, path, hdf_store)
-            else:
-                try:
-                    for subkey in include_from_model_in_hdf5[key]:
-                        if include_from_model_in_hdf5[key][subkey] is None:
-                            _save_model_property(getattr(getattr(self, key), subkey), subkey, os.path.join(path, key),
-                                                 hdf_store)
-                        elif callable(include_from_model_in_hdf5[key][subkey]):
-                            include_from_model_in_hdf5[key][subkey](subkey, os.path.join(path, key), hdf_store)
-                        else:
-                            logger.critical('Can not save %s', str(os.path.join(path, key, subkey)))
-                except:
-                    logger.critical('An error occurred while dumping %s to HDF.', str(os.path.join(path, key)))
-
-
-        hdf_store.flush()
-        if close_h5:
+        try:
```

Comment:
This is another nice point. Though, one thing that works better in the current way, is that the HDF file (if it is a file and not an HDFStore) isn't constantly opened and closed like it would be if we did pass the path to pandas, but I guess we can take this, since we are not talking about a really huge amount of IO operations.